Graylog2/graylog-plugin-integrations
# Upgrade to latest Kinesis Client version

## Overview
Upgrade to latest Kinesis Client version v2.2.1.

The upgrade to the latest [Kinesis Client Library](https://github.com/awslabs/amazon-kinesis-client) version was successful. State management (starting/stopping) and throttling all work well. 

This PR is dependent on the port of the older version of the Kinesis Transport consumer in #140. I propose that we merge #140 straightaway, and do all of the review here.

**Throttling notes**: Throttling appears to work much better in the new Kinesis Client Version 2.x. In this version, apparently, we can block in the `KinesisShardProcessorFactory.KinesisShardProcessor.processRecords()` method indefinitely without affecting the health of the consumer. My testing shows this working well so far. Therefore, the tricky consumer shutdown logic has been removed.

## General testing steps
1) Go to `/integrations/aws/cloudwatch` and complete the setup steps to add an AWS CloudWatch/Kinesis integration.  
2) Verify that the input starts up and pulls messages from a Kinesis stream in real time. 
3) Start/stop the input. Verify that the reading of messages actually stops and starts, and that no apparent duplicate/double messages are received.
4) Test throttling support. You can set your Graylog journal to a smaller size (by setting the `message_journal_max_size` [server.conf](https://docs.graylog.org/en/3.0/pages/configuration/server.conf.html?highlight=journal) setting to a smaller value) and reading messages from a very full Kinesis stream. You can use the `dtorrey-test-flowlogs` Kinesis stream in eu-west-1 for this (no need to set one up yourself).
